app/utils.py

```diff
@@ -92,14 +92,33 @@ def json_only_has_required_keys(data, keys):
         abort(400, "Invalid JSON must only have {} keys".format(keys))
 
 
-def drop_foreign_fields(json_object, list_of_keys):
-    json_object = json_object.copy()
-    for key in list_of_keys:
-        json_object.pop(key, None)
-
+def drop_foreign_fields(json_object, list_of_keys, recurse=False):
```

Comment:
I'm not sure if there's a need for the nested drop_foreign_fields.

For one, we don't have many nested questions we accept in writes. Contact information might be the only one, and it should've been a separate endpoint to create  contact information objects for a supplier to begin with. The only reason it's nested was to save time when were rolling this out.

And the way `drop_foreign_fields` is used everywhere else doesn't really allow it to be safely applied to nested objects. `list_of_keys` is a collection of keys we don't want to store for the root dictionary, but it's not necessarily the same for the nested objects. I believe the way we're approaching this now is calling this function from the model method, which allows us to list the keys we want removed once for each model.